blog/2023-02-04-mysql-slack-blog/index.md

```diff
@@ -0,0 +1,246 @@
+  -v ${PWD}:/vanus-connect/data \
+  --name source-mysql public.ecr.aws/vanus/connector/source-mysql &
+```
+
+12. Create an app on Slack. Visit Slack Website
+
+![slack ](img/clickcreate.gif)
+
+13. Click 'create new app' and from scratch
+
+![slack ](img/scratch.gif)
+
+14. Set the bot name as "Vanus-MySQL"
+15. Chose your workspace
+
+![slack ](img/entername.gif)
+
+16. Click on permissions in the central menu.
+
+![slack ](img/permission.gif)
+
+17. Set bot permissions  'Add OAuth Scope' chat:write and chat:write.public.
+
+![slack ](img/scopes.gif)
+
+18. Install the bot to your workspace.
+
+![slack ](img/installchannel.gif)
+
+19. Keep your Bot User OAuth Token in OAuth & Permissions.
+
+![slack ](img/token.gif)
+
+20. Create a yml file named slack-sink.yml in the playground with the following command:
+
+```shell
+cat << EOF > slack-sink.yml
+apiVersion: v1
+kind: Service
+metadata:
+  name: sink-slack
+  namespace: vanus
+spec:
+  selector:
+    app: sink-slack
+  type: ClusterIP
+  ports:
+    - port: 8080
+      name: sink-slack
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sink-slack
+  namespace: vanus
+data:
+  config.yml: |-
+    default: "test-app1"
+    slack:
+      - app_name: "test-app1"
+        token: "xoxb-429"
+        default_channel: "#general"
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sink-slack
+  namespace: vanus
+  labels:
+    app: sink-slack
+spec:
+  selector:
+    matchLabels:
+      app: sink-slack
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sink-slack
+    spec:
+      containers:
+        - name: sink-slack
+          image: public.ecr.aws/vanus/connector/sink-slack
+          resources:
+            requests:
+              memory: "128Mi"
+              cpu: "100m"
+            limits:
+              memory: "512Mi"
+              cpu: "500m"
+          imagePullPolicy: Always
+          ports:
+            - name: http
+              containerPort: 8080
+          volumeMounts:
+            - name: config
+              mountPath: /vanus-connect/config
+      volumes:
+        - name: config
+          configMap:
+            name: sink-slack
+EOF
+```
+
+21. Update the settings: `Channel` and `token`
+
+```shell
+vi slack-sink.yml
+```
+![slack yml](img/ymlupdateslack.gif)
+
+22. Create a subscription.
+
+    You can define your variable according to your own table structure or follow my structure below.
+    For example, username will be defined this way -> "user": "$.data.username", note that you need to add data. before your column.
+
+```mysql
+# My structure ->
+CREATE database Name;
+CREATE TABLE IF NOT EXISTS dataBaseName.tableName
+(
+  `id` int NOT NULL,
+  `username` varchar(100) NOT NULL,
+  `email` varchar(100) NOT NULL,
+  `password` varchar(100) NOT NULL,
+`subscription_plan` varchar(100) NOT NULL,
+`phone_number` varchar(100) NOT NULL,
+
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB;
+```
+
+```shell
+Create a subscription ->
```

Comment:
Create a subscription should not be part of the code

Reply:
I have a reason for doing this

Reply:
It throws errors on Playground

Reply:
Add # to make it a comment